utilities/replace_versions.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import re
+
+version_update_marker = re.compile(r'\{x-version-update:([^:]+):([^}]+)\}')
+version_update_start_marker = re.compile(r'\{x-version-update-start:([^:]+):([^}]+)\}')
+version_update_end_marker = re.compile(r'\{x-version-update-end\}')
+version_regex_str = r'\d+\.\d+\.\d+(?:-\w+)?(?:-\w+)?'
+version_regex = re.compile(version_regex_str)
+
+class CodeModule:
+    def __init__(self, module_str):
+        (name, released, current) = module_str.split(':')
+        self.name = name
+        self.released = released
+        self.current = current.strip()
+
+def replace_versions(version_map, target):
+    newlines = []
+    repl_open = False
+    repl_thisline = False
+    with open(target) as f:
+        for line in f:
+            repl_thisline = repl_open
+            match = version_update_marker.search(line)
+            if match:
+                module_name = match.group(1)
+                version_type = match.group(2)
+                repl_thisline = True
+            else:
+                match = version_update_start_marker.search(line)
+                if match:
+                    module_name = match.group(1)
+                    version_type = match.group(2)
+                    repl_open = True
+                    repl_thisline = True
+                else:
+                    match = version_update_end_marker.search(line)
+                    if match:
+                        repl_open = False
```

Comment:
In python you can do it like this (and I think community preferes this way):
```python
repl_open, repl_thisline = False, False
```

This applies to all similar blocks of initialization, and allows making the script substantially shorter.

Reply:
I'll make this change